sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala

```diff
@@ -96,37 +96,53 @@ object EvaluatePython {
       case c: Short => c.toByte
       case c: Int => c.toByte
       case c: Long => c.toByte
+      case c: Float => c.toByte
+      case c: Double => c.toByte
     }
 
     case ShortType => (obj: Any) => nullSafeConvert(obj) {
       case c: Byte => c.toShort
       case c: Short => c
       case c: Int => c.toShort
       case c: Long => c.toShort
+      case c: Float => c.toShort
+      case c: Double => c.toShort
```

Comment:
It seems odd that we would want to silently convert a double to a short. Are we sure that is the correct behavior?